src/main/java/org/apache/sling/jcr/repoinit/impl/AclUtil.java

```diff
@@ -291,6 +328,42 @@ public static void setPrincipalAcl(Session session, String principalName, Collec
         }
     }
 
+    public static void removePrincipalEntries(Session session, String principalName, Collection<AclLine> lines) throws RepositoryException {
+        final JackrabbitAccessControlManager acMgr = getJACM(session);
+        Principal principal = AccessControlUtils.getPrincipal(session, principalName);
+        if (principal == null) {
+            // due to transient nature of the repo-init the principal lookup may not succeed if completed through query
+            // -> save transient changes and retry principal lookup
+            session.save();
+            principal = AccessControlUtils.getPrincipal(session, principalName);
+            checkState(principal != null, "Principal not found: " + principalName);
+        }
+
+        final PrincipalAccessControlList acl = getPrincipalAccessControlList(acMgr, principal, true);
+        boolean modified = false;
+        for (AclLine line : lines) {
+            List<String> jcrPaths = getJcrPaths(session, line.getProperty(PROP_PATHS));
+            LocalRestrictions restr = createLocalRestrictions(line.getRestrictions(), acl, session);
+            List<String> privNames = line.getProperty(PROP_PRIVILEGES);
+            Privilege[] privs = AccessControlUtils.privilegesFromNames(session, privNames.toArray(new String[0]));
+            Predicate<PrincipalAccessControlList.Entry> predicate = entry -> {
+                if (!jcrPaths.contains(entry.getEffectivePath())) {
+                    return false;
+                }
+                LocalAccessControlEntry lace = new LocalAccessControlEntry(entry.getPrincipal(), privs, line.getAction()== AclLine.Action.ALLOW, restr);
+                return lace.isEqual(entry);
+            };
+            if (removePrincipalEntries(acl, principalName, predicate)) {
+                modified = true;
```

Comment:
Looking at the test coverage with a `-P jacoco-report` build reports that this line is not covered by tests, and that's also the case for the `if (modified)` statement below. Those lines look important, I suppose they should be tested.

Reply:
yes, i agree

Reply:
@bdelacretaz , in fact there was a test-method for that but i managed to garble the test during the many refactorings. should be fixed now. can you please confirm?